### ISSUE TITLE
Investigate datagrid `button` without tabbable element in datagrid toolbar

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4766]
Reported by @kristastarr in the [Datagrid review](https://github.ibm.com/CDAI-design/pal/issues/1575) from the accessibility audit.

```
 Component with “button” role does not have a tabbable element
```

This one could be a Carbon issue or an example/Storybook-specific one. Let’s investigate and either address or reopen.

![image](https://github.com/carbon-design-system/ibm-products/assets/1717398/08c7612a-9264-4b33-b809-a13882a70184)

> ### What to do
> After deciding that the component should be keyboard-interactive, use the tabindex attribute to include the component in the tab sequence
> 
> Otherwise, change the role to a non-widget or non-interactive type component so there is no expectation that the component should be operable with the keyboard
> 
> Rule ID: widget_tabbable_exists
> Reason ID: fail_no_tabbable
